src/memory/memory.ts

```diff
@@ -17,16 +17,15 @@ export class Memory {
   }
 
   incrementNumSegments(): void {
-    const { value: numSegments, error } = UnsignedInteger.toUint32(
-      this.numSegments + 1
-    );
-    if (error !== undefined) {
+    try {
+      this.numSegments += 1;
+      UnsignedInteger.ensureUint32(this.numSegments);
+    } catch (err) {
       throw new MemoryError(SegmentIncrementError);
     }
```

Comment:
why do we enforce numSegment to be uint32?

Reply:
I think because relocatable is numSegment: uint32, offset: u32